HPXMLtoOpenStudio/resources/hvac_sizing.rb

```diff
@@ -2678,8 +2678,9 @@ def self.gshp_hxbore_ft_per_ton(weather, hvac_cooling_ap, bore_spacing, pipe_r_v
     rtf_DesignMon_Heat = [0.25, (71.0 - weather.data.MonthlyAvgDrybulbs[0]) / @htd].max
     rtf_DesignMon_Cool = [0.25, (weather.data.MonthlyAvgDrybulbs[6] - 76.0) / @ctd].max
 
-    nom_length_heat = (1.0 - hvac_cooling_ap.heat_rated_eirs[0]) * (r_value_bore + r_value_ground * rtf_DesignMon_Heat) / (weather.data.AnnualAvgDrybulb - (2.0 * hvac_cooling_ap.design_hw - hvac_cooling_ap.design_delta_t) / 2.0) * UnitConversions.convert(1.0, 'ton', 'Btu/hr')
-    nom_length_cool = (1.0 + hvac_cooling_ap.cool_rated_eirs[0]) * (r_value_bore + r_value_ground * rtf_DesignMon_Cool) / ((2.0 * hvac_cooling_ap.design_chw + hvac_cooling_ap.design_delta_t) / 2.0 - weather.data.AnnualAvgDrybulb) * UnitConversions.convert(1.0, 'ton', 'Btu/hr')
+    ground_temp_f = WeatherProcess.get_undisturbed_ground_temperature(weather, climate_zone_iecc)
+    nom_length_heat = (1.0 - hvac_cooling_ap.heat_rated_eirs[0]) * (r_value_bore + r_value_ground * rtf_DesignMon_Heat) / (ground_temp_f - (2.0 * hvac_cooling_ap.design_hw - hvac_cooling_ap.design_delta_t) / 2.0) * UnitConversions.convert(1.0, 'ton', 'Btu/hr')
+    nom_length_cool = (1.0 + hvac_cooling_ap.cool_rated_eirs[0]) * (r_value_bore + r_value_ground * rtf_DesignMon_Cool) / ((2.0 * hvac_cooling_ap.design_chw + hvac_cooling_ap.design_delta_t) / 2.0 - ground_temp_f) * UnitConversions.convert(1.0, 'ton', 'Btu/hr')
```

Comment:
Both `nom_length_heat` and `nom_length_cool` use an annual average value. Would it be better to use the min and max (or perhaps winter and summer) monthly temperatures, respectively?

Reply:
@jmaguire1 : any thoughts here. I think you compared our sizing to various established L1-L4 methods.

Reply:
Let's resolve this question before merging -- do we want to make a change to address this now or not?

Reply:
I propose we use the `weather.data.GroundMonthlyTemps` array for the January and July values for `ground_temp_f` and then call this done for now. I will create an issue about how the L. Xing model could be used to update the `weather.data.GroundMonthlyTemps` array.

Reply:
Denver looks like this:
```
0 | 45.59726
1 | 42.4099
2 | 42.15465
3 | 43.66721
4 | 49.73931
5 | 55.78311
6 | 61.02277
7 | 64.34197
8 | 64.62395
9 | 61.92434
10 | 56.8183
11 | 50.95989
```
So use temperatures corresponding to 0 and 6? And not min and max? (Min and max might also be easier for southern hemisphere.)



</body>

</html>

Reply:
I think January/July would be better than min/max, since the design loads will be calculated using the months with the hottest/coldest outdoor air temperatures. You want to know how much borehole depth you need _coincident with when those design loads occur_.

Make sure to handle locations in the southern hemisphere correctly too.

Reply:
I'm on board with this (and I agree with Scott that Jan/July might be better than min/max, since that's consistent with a lot of our other design calculations).

@afontani: Do you remember what depth you used as an input to develop your correlation? The [Xing model](https://bigladdersoftware.com/epx/docs/8-7/engineering-reference/undisturbed-ground-temperature-model-xing.html) is a function of time and depth. I'm a little surprised how much of a seasonal swing we're seeing in the ground temperature here since we're generally talking about the temperature > 100 ft underground, I would have expected it to be a little more stable than what Joe is showing.

Reply:
Based on these changes, total bore lengths are increasing by 25-50% in our tests. That's a lot.

![image](https://github.com/NREL/OpenStudio-HPXML/assets/5861765/5a58f56d-107f-48c3-8e43-8f6fbff9332e)

@jmaguire1's comment makes me reconsider here. Indeed our monthly ground temperatures represent shallow temperatures primarily used for foundation heat transfer modeling. If we should be using deep ground temperatures here, then using the annual average is probably (?) better than what we've changed to, though we could look at the Xing data to confirm. (If we could develop monthly ground temperatures for deeper depths, that'd be the best possible outcome of course, but that would take more work.)